docs/tutorial_basics.rst

```diff
@@ -0,0 +1,1087 @@
+
+Now it's time to run our first test:
+
+.. code:: bash
+
+   ./bin/reframe -c tutorials/basics/hello/hello1.py -r
+
+
+.. code-block:: none
+
+   [ReFrame Setup]
+     version:           3.1-dev0 (rev: 986c3505)
+     command:           './bin/reframe -c tutorials/basics/hello/hello1.py -r'
+     launched by:       user@tresa.local
+     working directory: '/Users/user/reframe'
+     settings file:     '<builtin>'
+     check search path: '/Users/user/reframe/tutorials/basics/hello/hello1.py'
+     stage directory:   '/Users/user/reframe/stage'
+     output directory:  '/Users/user/reframe/output'
+
+   [==========] Running 1 check(s)
+   [==========] Started on Sat Jun 20 09:44:52 2020
+
+   [----------] started processing HelloTest (HelloTest)
+   [ RUN      ] HelloTest on generic:default using builtin
+   [----------] finished processing HelloTest (HelloTest)
+
+   [----------] waiting for spawned checks to finish
+   [       OK ] (1/1) HelloTest on generic:default using builtin [compile: 0.735s run: 0.505s total: 1.272s]
+   [----------] all spawned checks have finished
+
+   [  PASSED  ] Ran 1 test case(s) from 1 check(s) (0 failure(s))
+   [==========] Finished on Sat Jun 20 09:44:53 2020
+
+
+Perfect! We have verified that we have a functioning C compiler in our system.
+
+When ReFrame runs a test, it copies all its resources to a stage directory and performs all test-related operations (compilation, run, sanity checking etc.) from that directory.
+On successful outcome of the test, the stage directory is removed by default, but interesting files are copied to an output directory for archiving and later inspection.
+The prefixes of these directories are printed in the first section of the output.
+Let's inspect what files ReFrame produced for this test:
+
+.. code-block:: bash
+
+   ls output/generic/default/builtin/HelloTest/
+
+.. code-block:: none
+
+   rfm_HelloTest_build.err rfm_HelloTest_build.sh  rfm_HelloTest_job.out
+   rfm_HelloTest_build.out rfm_HelloTest_job.err   rfm_HelloTest_job.sh
+
+ReFrame stores in the output directory of the test the build and run scripts it generated for building and running the code along with their standard output and error.
+All these files are prefixed with ``rfm_``.
+
+
+More of "Hello, World!"
+-----------------------
+
+We want to extend our test and run a C++ "Hello, World!" as well.
+We could simply copy paste the ``hello1.py`` and change the source file extension to refer to the C++ source code.
+But this duplication is something that we generally want to avoid.
+ReFrame allows you to avoid this in several ways but the most compact is to define the new test as follows:
+
+
+.. literalinclude:: ../tutorials/basics/hello/hello2.py
+   :lines: 6-
+
+
+This exactly the same test as the ``hello1.py`` except that it is decorated with the :func:`@parameterized_test <reframe.core.decorators.parameterized_test>` decorator instead of the :func:`@simple_test <reframe.core.decorators.simple_test>`.
```

Comment:
```suggestion
This is exactly the same test as the ``hello1.py`` except that it is decorated with the :func:`@parameterized_test <reframe.core.decorators.parameterized_test>` decorator instead of the :func:`@simple_test <reframe.core.decorators.simple_test>`.
```